components/shared_code/src/shared_data_model/parameters.py

```diff
@@ -151,6 +151,15 @@ class Branch(StringParameter):
     metrics: list[str] = ["source_up_to_dateness"]
 
 
+class Branches(MultipleChoiceWithAdditionParameter):
+    """Branches parameter."""
+
+    name: str = "Branches (regular expressions or branch names)"
```

Comment:
Does this mean it is an _inclusive_ filter parameter?

Reply:
What about this `TargetBranchesToInclude` on line 171?

Reply:
I'm not sure what you mean with inclusive filter (if a filter includes something it also excludes things, right?). Anyway, if the user does not add any branches to the `Branches` the collector considers pipelines on all branches (as signalled by the placeholder being `all branches`. As soon as the user adds one or more branches, the collector only looks at pipelines for those branches.

Yeah,`TargetBranchesToInclude` is also a `MultipleChoiceWithAdditionParameter` parameter for branches, but all attributes are different, so I don't see a way to get rid of it.